diff -ruN icecast-2.0.0.orig/conf/icecast.xml.in icecast-2.0.0/conf/icecast.xml.in
--- icecast-2.0.0.orig/conf/icecast.xml.in	2004-01-08 15:13:43.000000000 +0100
+++ icecast-2.0.0/conf/icecast.xml.in	2004-01-09 21:42:49.000000000 +0100
@@ -108,11 +108,9 @@
 
     <security>
         <chroot>0</chroot>
-        <!--
         <changeowner>
             <user>nobody</user>
             <group>nogroup</group>
         </changeowner>
-        -->
     </security>
 </icecast>
